platform/android/app/src/main/kotlin/coop/polypoly/polypod/Authentication.kt

```diff
@@ -46,28 +46,28 @@ class Authentication {
 
         fun authenticate(
             activity: FragmentActivity,
-            newStatus: Boolean = false,
+            showAuthTexts: Boolean = false,
             authComplete: ((Boolean) -> Unit)
         ) {
             val isBiometricEnabled = Preferences.isBiometricEnabled(activity)
             if (!biometricsAvailable(activity) ||
-                (!newStatus && !isBiometricEnabled)
+                (!showAuthTexts && !isBiometricEnabled)
```

Comment:
`showAuthTexts` is highly UI specific, from the code it is not clear why it should be used to determine if user should be authenticated or not.

Reply:
I do find this `if` ambivalent anw, the `isBiometricEnabled` will be affected after this code so why to check the newStatus with its combination? I'd say to revisit this part and have a refactor here as there were edge cases marked by Lars that are uncovered up to now.